### PR TITLE
sci-mathematics/pari: new release + upstream patch

### PR DESCRIPTION
**Pull request by kiwifb**
Package-Manager: Portage-2.3.89, Repoman-2.3.20
Signed-off-by: François René Pierre Bissey <frp.bissey@gmail.com>

### CODE REVIEW

File: sci-mathematics/pari/pari-2.11.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit flag-o-matic toolchain-funcs
+
+DESCRIPTION="Computer-aided number theory C library and tools"
+HOMEPAGE="http://pari.math.u-bordeaux.fr/"
```
Comment (charIes17) [inline sci-mathematics/pari/pari-2.11.3.ebuild:9]: https://pari.math.u-bordeaux.fr/
maybe also for SRC_URI ?

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild:9]: Right, this should be done indeed. But the other ebuild already present should be touched too. But that would be a separate PR of course.

**Comment by orlitzky** [comment]
I've written in metadata.xml that you should let @kiwifb do whatever he wants here, he's been the real maintainer for a long time.

File: sci-mathematics/pari/pari-2.11.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+PATCHES=(
+	"${FILESDIR}/${PN}"-2.9.4-strip.patch
+	"${FILESDIR}/${PN}"-2.9.4-ppc.patch
+	"${FILESDIR}/${PN}"-2.11.2-no-automagic.patch
+	"${FILESDIR}/${PN}"-2.9.4-fltk-detection.patch
+	"${FILESDIR}/${PN}"-2.11.2-Makefile-LDFLAGS.patch
+	"${FILESDIR}/${PN}"-2.11.2-Makefile-docinstall.patch
+	"${FILESDIR}/${PN}"-2.11.3-incorrect_bnfisunit.patch
+	)
```
Comment (juippis) [inline sci-mathematics/pari/pari-2.11.3.ebuild:36]: Remove indent here please.

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild:36]: Fixing the indent is trivial. The latex problem less so. I spent quite some time upstreaming parallel building fix of the doc over the years, so I'll discard that possibility for now. The basic outline as type1 font is part of the texlive-basic ebuild and is part of the base install of texlive. But the problem is with generating the bitmap version.

Educated guess, do you have texlive-mathscience installed?

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild:36]: I don't understand the problem you are having with the doc. I happened to have lost the OS disk on one of my gentoo setup recently. So I have a fresh system built from scratch that didn't have any texlive or latex whatsoever to test with. After doing `emerge -uv virtual/latex-base` I was able to install pari-2.11.3 with the doc, no extra install steps.

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild:36]: These are the packages that have been installed by `virtual/latex-base`
```
[I] app-text/texlive-core (2019-r10@04/11/20): A complete TeX distribution
[I] dev-texlive/texlive-basic (2019-r1@04/11/20): TeXLive Essential programs and files
[I] dev-texlive/texlive-fontsrecommended (2019@04/11/20): TeXLive Recommended fonts
[I] dev-texlive/texlive-fontutils (2019@04/11/20): TeXLive Graphics and font utilities
[I] dev-texlive/texlive-latex (2019@04/11/20): TeXLive LaTeX fundamental packages
[I] dev-texlive/texlive-latexrecommended (2019-r2@04/11/20): TeXLive LaTeX recommended packages
[I] dev-tex/bibtexu (3.71_p20190410@04/11/20): 8-bit Implementation of BibTeX 0.99 with a Very Large Capacity
[I] dev-libs/kpathsea (6.3.1_p20190410(0/6.3.1)@04/11/20): Path searching library for TeX-related files
[I] app-text/dvipsk (5.999_p20190410@04/11/20): DVI-to-PostScript translator
[I] app-text/ps2pkm (1.8_p20190410@04/11/20): Tool that converts a PostScript type1 font into a corresponding TeX PK font
```
Can you check if you somehow miss one of them?

Comment (juippis) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: I had all of those installed, however my texlive-core is newer than yours (-r11). I couldn't build -r10, it failed to whoknows what. 

I tried enabling `xetex` for texlive-core but it didn't help. Getting the same error. Tried googling it too but I can't make much out of those answers. 

I use a clean stage3 image to test these pull requests, so chances are you've done something in your system to make it work, but it'd fail for ... most people I guess, unless they work with latex a lot. That's why I'd like to solve this issue here, so you/we won't get bug reports about it later.

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: Like I said I did a test with a system that was fresh and had 0 tex/latex stuff installed. I appreciate your concern and it is absolutely valid. I would have thought the essential packages where `texlive-basic` , `kpathsea` and `ps2pkm` but obviously there is something more to be done. I will try -r11 to see if I can reproduce the problem. After that, we may be looking to see if it needs a special font.

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: Can you provide a log for a serial (-j1) build?

Comment (juippis) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: https://cpy.pt/zbMWkmH4 here.

I found https://forums.gentoo.org/viewtopic-t-1030660-start-0.html but didn't try that yet.

Comment (juippis) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: Thanks for looking into this, let me know if you find something.

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: For the record, on my fresh system I don't have `ttf2pk2` which would be enabled by `texlive[truetype]`. Was a good try, though. I don't have `extra` on any of my systems, and none of the pulled packages are on my fresh system. Looking at the log now.

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: Do you have the file `/usr/share/texmf-dist/fonts/tfm/public/amsfonts/euler/eufm8.tfm`?

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: Did you do something to get tex/latex/metafont extra verbose? You have a lot of metafont chatter I just don't. I suspect your pdftex.map is broken or not up to date somehow.

Comment (juippis) [reply sci-mathematics/pari/pari-2.11.3.ebuild]:
```
 >: file /usr/share/texmf-dist/fonts/tfm/public/amsfonts/euler/eufm8.tfm
/usr/share/texmf-dist/fonts/tfm/public/amsfonts/euler/eufm8.tfm: shared library
```

To my knowledge I haven't done anything except selected default/linux/amd64/17.1 as profile and tried to emerge this. Although in order to debug this a bit, I did install texlive-2019-r1.

I do use `X` globally too, per 
https://ebuildtester.readthedocs.io/en/latest/#command-line-arguments

```
[ebuild   R    ] app-text/texlive-2019-r1::gentoo  USE="X png truetype xml -cjk -context -detex -dvi2tty -epspdf -extra -games -graphics -humanities -jadetex -luatex -metapost -music -pdfannotextractor -pstricks -publishers -science -tex4ht -texi2html -xetex -xindy"
[ebuild   R    ] app-text/texlive-core-2019-r11::gentoo  USE="X luajittex -cjk -doc -source -tk -xetex*" 0 KiB
[ebuild  N     ] sci-mathematics/pari-2.11.2:0/6::gentoo  USE="X -data -doc -fltk -gmp -test -threads" 4,330 KiB
```
(Note that I'd enable `doc` on pari to test it)
Do you have `xetex` enabled on both texlive packages?

Comment (juippis) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: Side question: Do you know what handles pdftex.map in Gentoo? Maybe the fault is found there.

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: I have `xetex` on one system but not on the "fresh" one.  I suspect one of the texlive/latex eclass is responsible for updating pdftex.map but I have to check that.
Another test
```
$ grep eufm8 /usr/share/texmf-dist/fonts/map/pdftex/updmap/pdftex.map
eufm8 EUFM7 <eufm7.pfb
```
Do you get that too?

Comment (juippis) [reply sci-mathematics/pari/pari-2.11.3.ebuild]:
```
 >: grep eufm8 /usr/share/texmf-dist/fonts/map/pdftex/updmap/pdftex.map
eufm8 EUFM7 <eufm7.pfb
```
Looks the same at least.

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: You wouldn't happen to have some stuff in `~/.texlive*` or even `.local/tmp`? https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=856020

Comment (juippis) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: Neither exists :)

Need to step off for few hours to do paid work, gonna return today if you have some ideas. 

Honestly this looks hard, I think at the end of the day I'm gonna test running the command from Gentoo forums we saw earlier, and if that doesn't fix it, just merge and wait for bug reports.

Comment (kiwifb) [reply sci-mathematics/pari/pari-2.11.3.ebuild]: For the record: both `latex-package.eclass` and `texlive-common.eclass` run `texmf-update` which should deal with the fonts.
On the other hand I don't see how you could have a problem with 2.11.3 and not 2.11.2. Have you tried to install 2.11.2, which is the current in tree version, with `USE=doc`?

**Comment by juippis** [comment]
15f489e9fbc977dc72d9bdcabc8b19dce3e2aabf

**Pull request closed without merging**